pkg/relayer/proxy/server_builder.go

```diff
@@ -0,0 +1,57 @@
+package proxy
+
+import (
+	"context"
+
+	sharedtypes "pocket/x/shared/types"
+	suppliertypes "pocket/x/supplier/types"
+)
+
+type RelayServersMap map[string][]RelayServer
```

Comment:
NIT: wdyt about something like this to be more explicit about what the key is expected to be?:

```suggestion
type (
  serviceId = string
  RelayServersMap map[serviceId][]RelayServer
)
```